lib/spack/spack/new_installer.py

```diff
@@ -1294,6 +1293,17 @@ def __init__(
         self.running_builds: Dict[int, ChildInfo] = {}
         self.build_status = BuildStatus(len(self.build_graph.nodes))
         self.jobs = spack.config.determine_number_of_jobs(parallel=True)
+        if concurrent_packages is None:
+            # If no command line override, get from config, and treat 1 as unlimited, because
+            # the 1 value is the historical default of the old installer where concurrency was
+            # disabled by default.
+            concurrent_packages_config = spack.config.get("config:concurrent_packages", 1)
+            if concurrent_packages_config == 1:
+                self.capacity = sys.maxsize
```

Comment:
nitpick: we called it `concurrent_packages` on the CLI after much debate... why not call the instance variable that?